docs: Add release notes for Metals 1.5.2 (#7319)
scalameta/metals: website/blog/2025-03-20-strontium.md

---
authors: tgodzik
title: Metals v1.5.2 - Strontium
---

We're happy to announce the release of Metals v1.5.2, which continues to improve
overall user experience and stability of Metals. This release includes a number
of new features, bug fixes and improvements. Especially worth mentioning is the

> fix for the Scala 2 issue, which would break most interactive features if a type
> was used that was defined in a package object.

Reviewer: ```suggestion
fix for the Scala 2 issue, which previously caused most interactive features to break when using a type defined in a package object.
```

- [Remove Ammonite script support](#remove-ammonite-script-support)

## Deduplicate compile requests

In previous versions of Metals, we would sometimes send multiple compile
requests for the same file expecting most of them to be fast. This worked
reliably well in Bloop, but not so much in a lot of other build servers such as
sbt.

Thanks to [kasiaMarek](https://github.com/kasiaMarek) we will try not to send
multiple requests if we know a file hasn't changed between them. This should
help in a number of cases and make the experience more reliable.

## Add exact location for the test failure

Previously, when running tests in Metals with VS Code Test Explorer, the
location of the test failure was not shown, only the start of the test case.
Now, we will correctly show the exact location of the test failure, which should
make it easier to navigate to the failing test.

![test-loc](https://i.imgur.com/JuIES78.gif)

## Convert sbt style deps on paste in for scala-cli

Thanks to [majk-p](https://github.com/majk-p) any sbt style dependencies that
you paste into a Scala CLI file after `//> using dep` will be automatically

> converted to Scala CLI style dependencies. This was previously only supported in
> a code action, but turns out to be a very useful feature that should be
> available in more places.

Reviewer: ```suggestion
converted to Scala CLI style dependencies. This was previously only supported in
a code action, but turns out to be useful enough to apply the rewrite automatically on paste.
```

![paste-dep](https://i.imgur.com/6BNvmmO.gif)

## Add test cases discovery for TestNG

Thanks to [kasiaMarek](https://github.com/kasiaMarek) it's now possible to run
TestNG tests inside Metals, which was previously not possible.

This would mean following test cases will now show lenses or test explorer icon
next to them:

```scala
import org.testng.annotations.Test

class TestNG {
    @Test
    def testOK(): Unit = {
        assert(true)
    }
}
```

## Improvements to automatic imports

Whenever importing a particular symbol, Metals would previously suggest all the
possible symbol with the expected name. However, in a lot of cases, those symbol

> might already have a method being invoked on them, which means that only symbol
> containing that method should only be suggested.

Reviewer: ```suggestion
might already have a method being invoked on them, which means that only symbols
containing that method should be suggested.
```

For example in the following code:

```scala
object O{
  Future.successful(1)
}
```

we don't need to suggest java.concurrent.Future, but only
scala.concurrent.Future since only the latter contains the `successful` method.
If none of the symbols contains that specific method, we will revert to showing
all symbols. Let us know if this is a helpful behaviour or if you see additional
cases where this could be improved.

## Remove Ammonite script support

Some time ago we have asked in
[discussions](https://github.com/scalameta/metals/discussions/6522) in our
Github repository about the possibility of removing support for Ammonite
scripts. We have not received any overwhelming feedback to keep it, so we have
decided to remove it. If you have any concerns or questions about this change,
please let us know.

The removal was directly prompted by the fact that there were serious issues
with Ammonite support with no users reporting them or anyone from maintainers
being aware of them. Which points to the fact that the feature is not widely
used.

This will also help to to ease the maintenance burden and to focus more on

> stability of Metals. Scala Scripts are still supported via
> [Scala CLI](https://scala-cli.virtuslab.org/), which we believe is overall a
> better experience and is the default Scala runner for Scala 3.

Reviewer: ```suggestion
stability of Metals. Scala Scripts still are and will be supported via
[Scala CLI](https://scala-cli.virtuslab.org/), so for anyone using Ammonite we suggest switching to Scala CLI. Scala CLI is the default Scala runner for Scala 3 and we believe it offers overall better experience. If you see any useful Ammonite feature missing from Scala CLI don't hesitate to start a discussion in the Scala CLI repository.
```
